inspirehep/modules/orcid/builder.py

```diff
@@ -0,0 +1,313 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2014-2017 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""
+Builds an ORCID work record.
+"""
+
+from __future__ import absolute_import, division, print_function
+from lxml import etree
+from lxml.builder import ElementMaker
+from six import text_type
+
+_NAMESPACES = {
+    'work': 'http://www.orcid.org/ns/work',
+    'common': 'http://www.orcid.org/ns/common'
+}
+
+_WORK = ElementMaker(namespace=_NAMESPACES['work'], nsmap=_NAMESPACES)
+_COMMON = ElementMaker(namespace=_NAMESPACES['common'], nsmap=_NAMESPACES)
+
+_ELEMENT_MAKERS = {
+    'work': _WORK,
+    'common': _COMMON
+}
+
+
+class OrcidBuilder(object):
+    """
+    Class used to build ORCID-compatible work records in JSON.
+    """
+    def __init__(self):
+        self.record = _WORK.work()
+
+    def get_xml(self):
+        """Get an XML record.
+
+        Returns:
+            lxml.etree._Element: ORCID work record compatible with API v2.0
+        """
+        return self.record
+
+    def __str__(self):
+        """Get a string-serialized XML respresentation of a record.
+
+        Returns:
+            string: ORCID work record as an XML string
+        """
+        return etree.tostring(self.get_xml())
+
+    def add_title(self, title, subtitle=None, translated_title=None):
+        """Set a title of the work, and optionaly a subtitle.
+
+        Args:
+            title (string): title of the work
+            subtitle (string): subtitle of the work
+            translated_title (Tuple[string, string]): tuple consiting of the translated title and its language code
+        """
+        title_field = _WORK.title(_COMMON.title(title))
+
+        if subtitle:
```

Comment:
Do we support the case when we want subtitle to be the empty string ''?
If that's the case this clause should be: `if subtitle is not None`

Reply:
no, as you can see in the [JSON schema](https://github.com/inspirehep/inspire-schemas/blob/master/inspire_schemas/records/elements/title.yml), it has to have length at least 1, that is, be non-empty. This is a general feature: all strings and arrays in the records have to be non-empty. Absence of a value should be represented as absence of the corresponding key.